File: MEDIUM/Min Swaps to Group 1s/solution.cpp
class Solution {
  public:
    int minSwaps(vector<int>& arr) {
        int n = arr.size();
        
        int k = 0;
        for(int x : arr) {
            if(x == 1) k++;
        }
        
        if(k == 0) return -1;
        
        int ones = 0;
        
        for(int i = 0; i < k; i++) {
            if(arr[i] == 1) ones++;
        }
        
        int maxOnes = ones;
        
        for(int i = k; i < n; i++) {
            if(arr[i] == 1) ones++;
            if(arr[i - k] == 1) ones--;
            
            maxOnes = max(maxOnes, ones);
        }
        
        return k - maxOnes;
    }
};

File: MEDIUM/Min Swaps to Group 1s/solution.py
class Solution:
     def minSwaps(self, arr):
        n = len(arr)
        ones = arr.count(1)
        if ones == 0:
            return -1
        if ones == n:
            return 0
        window = max_ones = sum(arr[:ones])
        for i in range(ones, n):
            window += arr[i] - arr[i - ones]
            if window > max_ones:
                max_ones = window
        return ones - max_ones